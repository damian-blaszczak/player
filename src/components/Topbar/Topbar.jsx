import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import routes from '../../routes.const';
import styled from 'styled-components';
import Title from '../Title/Title';
import CloseIcon from '@material-ui/icons/Close';
import MoreVertIcon from '@material-ui/icons/MoreVert';

const TopbarContainer = styled.div`
  background-color: #534054;
  display: flex;
  align-items: center;
  padding: 15px 20px;
`;

const whiteColorStyle = { color: '#fff' };

const Topbar = ({ title }) => (
  <TopbarContainer>
    <Link to={routes.intro}>
      <CloseIcon
        fontSize="large"
        style={whiteColorStyle}
      />
    </Link>
    <Title>{title}</Title>
    <MoreVertIcon
      fontSize="large"
      style={whiteColorStyle}
    />
  </TopbarContainer>
);

Topbar.propTypes = {
  title: PropTypes.string
};

export default Topbar;
