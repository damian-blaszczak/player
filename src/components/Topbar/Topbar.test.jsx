import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Topbar from './Topbar';
import { MemoryRouter as Router } from 'react-router-dom';

it('renders correctly', () => {
  const { container } = render(
    <Router>
      <Topbar />
    </Router>
    );

  expect(container).toMatchSnapshot();
});
