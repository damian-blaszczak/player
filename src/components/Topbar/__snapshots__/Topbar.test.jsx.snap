// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-AxirZ fyWRYB"
  >
    <a
      href="/"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
        focusable="false"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </a>
    <h2
      class="sc-AxjAm ijAwJO"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      style="color: rgb(255, 255, 255);"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </div>
</div>
`;
